app/services/PensionSessionService.scala

```diff
@@ -118,4 +119,32 @@ class PensionSessionService @Inject() (pensionUserDataRepository: PensionsUserDa
       case Left(error: DatabaseError) => Left(error)
       case Right(_)                   => Right(())
     }
+
+  def mergePriorDataToSession(
+      taxYear: Int,
+      user: User,
+      renderView: (Int, PensionsCYAModel, Option[AllPensionsData]) => Result
+  )(implicit request: Request[_], hc: HeaderCarrier, clock: Clock): Future[Result] =
+    getAndHandle(taxYear, user) { (sessionData, priorData) =>
+      createOrUpdateSessionIfNeeded(sessionData, priorData, taxYear, user, renderView)
+    }
+
+  private def createOrUpdateSessionIfNeeded(
+      pensionsUserData: Option[PensionsUserData],
```

Comment:
Minor thing, but I think it helps with readability to rename `pensionsUserData` to `sessionData` so it is obvious at first glance what it is.

Reply:
done